src/js/productList.mjs

```diff
@@ -3,27 +3,47 @@ import { renderListWithTemplate } from "./utils.mjs";
 
 function productCardTemplate(product) {
   return `<li class="product-card">
-            <a href="product_pages/index.html?product=${product.Id}">
-              <img
-                src="${product.Image}"
-                alt="Image of ${product.Name}"
-              />
-              <h3 class="card__brand">${product.Brand.Name}</h3>
-              <h2 class="card__name">${product.Name}</h2>
-              <p class="product-card__price">$${product.FinalPrice}</p></a>
-          </li>`; 
+      <a href="product_pages/index.html?product=${product.Id}">
+      <img
+      src="${product.Image}"
+      alt="Image of ${product.Name}"
+      />
+      <h3 class="card__brand">${product.Brand.Name}</h3>
+      <h2 class="card__name">${product.Name}</h2>
+      <p class="product-card__price">$${product.FinalPrice}</p></a>
+      </li>`;
 }
 
-export default async function productList(selector = "", category = "") {
-  //TODO add something 
-  const itemListArea = document.getElementById(selector); 
-  const products = await getData(category);  
-  const inactiveProducts = ["989CG", "880RT"]; 
-  let productsTemp = products.filter(function (item) {
-    
-    return inactiveProducts.indexOf(item.Id) === -1; 
-  }); 
+async function handleErrorScr(product) {
+  return await fetch(product.Image).then((response) => {
+    console.debug(response);
+    if (response.status === 404) {
+      return false;
+    } else {
+      return true;
+    }
+  });
+}
+
+async function filterProductsWithoutImage(products) {
```

Comment:
Filtering via async function is tricky. We first need to make sure all our promises are resolved before moving on (otherwise the filter won't work because it'll just be a bunch of truthy promises). So first we map all products to a bunch of promises and then use `Promise.all` to make sure they all resolve and then work from there.